lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -1101,6 +1101,78 @@ ReadOp::removeBlockingUses(const MemorySlot &slot,
   return DeletionKind::Delete;
 }
 
+//===----------------------------------------------------------------------===//
+// PowSOp
+//===----------------------------------------------------------------------===//
+
+static OpFoldResult powCommonFolding(MLIRContext *ctxt, Attribute lhs,
+                                     Attribute rhs) {
+  auto lhsValue = dyn_cast_or_null<FVIntegerAttr>(lhs);
+  if (lhsValue && !lhsValue.getValue().hasUnknown() &&
+      lhsValue.getValue().toAPInt(false) == 1)
```

Comment:
I think `lhs.getValue() == 1` will do the same thing. FVInt has the `==` operator overloaded, which will check if the FVInt has no unknowns and whether the value matches.